### PR TITLE
Pc/azure ml pipelines

### PR DESCRIPTION
**Pull request by Blaizzy**
# Description

Include a summary of the changes and the related issue.

__Related to:__ `(ClickUp/JIRA task name)`

__Any expected test failures?__

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [x] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [x] Refactored code ([sourcery](https://sourcery.ai/))
- [x] Tested code locally
- [x] Precommit installed and run before pushing changes
- [ ] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [x] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS:
- Python version:
- Neptune version:
- Affected libraries with version:

### CODE REVIEW

**Review by SiddhantSadangi — APPROVED** [review]

**Comment by SiddhantSadangi** [comment]
@normandy7 - this is already merged as it was time-critical for a workshop, but you can still leave your suggestions and have this as a reference for the upcoming docs page on AzureML support

File: integrations-and-supported-tools/azure-ml/build_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+AZURE_SUBSCRIPTION_ID = "<YOUR SUBSCRIPTION ID>"
+AZUREML_RESOURCE_GROUP_NAME = "<YOUR RESOURCE GROUP NAME>"
+AZUREML_WORKSPACE_NAME = "<YOUR WORKSPACE NAME>"
+
+
+def compose_pipeline(
+    compute_target="cpu-cluster",
+    custom_env_name="neptune-example",
+    custom_env_version="2",
+    neptune_project="common/project-time-series-forecasting",
```
Comment (normandy7) [inline integrations-and-supported-tools/azure-ml/build_pipeline.py:27]: Is this supposed to be hardcoded, or could it use `NEPTUNE_PROJECT`?

File: integrations-and-supported-tools/azure-ml/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# AzureML how-to
```
Comment (normandy7) [inline integrations-and-supported-tools/azure-ml/README.md:1]: We should here use _Azure Machine Learning_, as "AzureML" is not an official term.

File: integrations-and-supported-tools/azure-ml/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+This project is an example integration between Azure DevOps and Azure ML services with Neptune.
+
+## Environment preparation
+
+### AzureML
```
Comment (normandy7) [inline integrations-and-supported-tools/azure-ml/README.md:7]: Same as above; we should use _Azure Machine Learning_.

File: integrations-and-supported-tools/azure-ml/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+To run the example, first create a compute cluster and a custom environment in AzureML environment. You can do that by executing the `./dependencies/build_compute_cluster.py` and `./dependencies/build_environment.py` scripts.
```
Comment (normandy7) [inline integrations-and-supported-tools/azure-ml/README.md:9]: "in your Azure ML environment"?

File: integrations-and-supported-tools/azure-ml/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+```
+AZURE_SUBSCRIPTION_ID = "<YOUR SUBSCRIPTION ID>"
+AZUREML_RESOURCE_GROUP_NAME = "<YOUR RESOURCE GROUP NAME>"
+AZUREML_WORKSPACE_NAME = "<YOUR WORKSPACE NAME>"
+```
+with values representing your environment.
+
+### Azure DevOps
+
+For Azure DevOps Pipelines to be able to successfully create and execute AzureML Pipelines, create the following secrets as per `./azure-ci/azure-pipelines.yaml` in your Azure DevOps Pipeline via the UI:
```
Comment (normandy7) [inline integrations-and-supported-tools/azure-ml/README.md:22]: "Azure ML"

File: integrations-and-supported-tools/azure-ml/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+```
+AZURE_TENANT_ID: $(tenant)
+AZURE_CLIENT_ID: $(client)
+AZURE_CLIENT_SECRET: $(secret)
+NEPTUNE_API_TOKEN: $(neptune-sa-token)
+```
+
+## The example
+The example is focused around creation of an Azure DevOps CI/CD pipeline that would be able to test the AzureML Pipeline and then deploy it for operational purposes. The following picture shows the resulting AzureML pipeline:
```
Comment (normandy7) [inline integrations-and-supported-tools/azure-ml/README.md:32]: "Azure ML"

**Review by normandy7 — COMMENTED** [review]
Just some notes about how to spell "Azure Machine Learning".

File: integrations-and-supported-tools/azure-ml/build_pipeline.py (hunk 1 above)
Comment (SiddhantSadangi) [reply integrations-and-supported-tools/azure-ml/build_pipeline.py:27]: Should use `NEPTUNE_PROJECT`

**Pull request merged**